src/data_source/storage/sql.rs

```diff
@@ -1598,10 +1595,10 @@ where
         let rows = self
             .query(
                 &format!(
-                    "SELECT DISTINCT ON (pos) index
+                    "SELECT DISTINCT ON (path) index
                     FROM {state_type}
                     WHERE created <= $1 AND index IS NOT NULL
-                    ORDER BY pos, created DESC;"
+                    ORDER BY path, created DESC;"
```

Comment:
Does this `ORDER BY path` still do the right thing without using ltree? Tbh we've gotten rid of the need for this and `load_snapshot` in the sequencer now. We could delete these two functions and thus remove a lot of maintenance burden in this repo. If we ever want to add them back, we have the git history

Reply:
The idea of having a randomized test might be useful. We could keep that but just test `get_path` in it instead of `get_snapshot`

Reply:
I'll leave it up to you, if you want to keep this and it still works with the new way of doing it then I'm fine leaving it in

Reply:
I think yes ORDER BY is working as expected for arrays too. For LTREE it is created as string 
![image](https://github.com/EspressoSystems/hotshot-query-service/assets/45506001/041fd4ac-d4ea-4a29-86b0-02480f8f36a8)